library/src/google-maps-provider.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {useState, useEffect, PropsWithChildren} from 'react';
+
+// https://developers.google.com/maps/documentation/javascript/url-params
+export interface GoogleMapsAPIUrlParameters {
+  libraries?: string[];
+  language?: string;
+  region?: string;
+  version?: string;
+  authReferrerPolicy?: string;
+}
+
+export interface GoogleMapsProviderProps extends GoogleMapsAPIUrlParameters {
+  googleMapsAPIKey: string;
+  mapContainer?: HTMLElement | null;
+  mapOptions?: google.maps.MapOptions;
+  onLoadScript?: () => void;
+  onLoadMap?: (map: google.maps.Map) => void;
+}
+
+/**
+ * The Google Maps map context
+ */
+export const GoogleMapsMapContext = React.createContext<
+  google.maps.Map | undefined
+>(undefined);
+
+/**
+ * The global Google Maps provider
+ */
+export const GoogleMapsProvider: React.FunctionComponent<
+  PropsWithChildren<GoogleMapsProviderProps>
+> = props => {
+  const {
+    children,
+    googleMapsAPIKey,
+    mapContainer,
+    mapOptions,
+    libraries,
+    language,
+    region,
+    version,
+    authReferrerPolicy,
+    onLoadScript,
+    onLoadMap
+  } = props;
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [map, setMap] = useState<google.maps.Map>();
+
+  // Load Google Maps API
+  useEffect(() => {
+    if (google?.maps) {
+      setIsLoading(false);
+      return () => {};
+    }
+
+    const scriptTag = document.createElement('script');
+    const defaultLanguage = navigator.language.slice(0, 2);
+    const defaultRegion = navigator.language.slice(3, 5);
+
+    /* eslint-disable camelcase */
+    const params = new URLSearchParams({
+      key: googleMapsAPIKey,
+      language: language || defaultLanguage,
+      region: region || defaultRegion,
+      ...(libraries?.length && {libraries: libraries.join(',')}),
+      ...(version && {version}),
+      ...(authReferrerPolicy && {auth_referrer_policy: authReferrerPolicy})
+    });
+    /* eslint-enable camelcase */
+
+    scriptTag.type = 'text/javascript';
+    scriptTag.src = `https://maps.googleapis.com/maps/api/js?${params.toString()}`;
+    scriptTag.onload = (): void => {
+      setIsLoading(false);
+      onLoadScript && onLoadScript();
+    };
+    document.getElementsByTagName('head')[0].appendChild(scriptTag);
+
+    // Remove all loaded Google Maps scripts
+    return () => {
+      document
```

Comment:
Why this way again and not an `id` and checking above? --> Generally just mean is still also necessary to do the 'destroyComplete' if we would just check for the id?

Reply:
Because I don't only want to remove the script tag I added above, but all script tags, that were additionally added by the Google Maps API.

Reply:
Okay, I was wondering if this is the reason, but wasn't sure. Then perfect 👍 

To keep all of this in a `destroy function` is not be wished here anymore? Might be a bit clearer what is happening for people looking into the code. Like having it at a first glance.

Reply:
I don't see the advantage of putting this in a separate function, but I'll try to make this clearer with code comments.